R/xml_nodes_to_lints.R

```diff
@@ -3,6 +3,13 @@
 #' Convenience function for converting nodes matched by XPath-based
 #'   linter logic into a [Lint()] object to return.
 #'
+#' @details
+#' The location XPaths, `column_number_xpath`, `range_start_xpath` and `range_end_xpath` are evaluated using
+#' [xml2::xml_find_num()] and will usually be of the form
+#'   "number(./relative/xpath)"
+#' Note that the location line number cannot be changed and lints spanning multiple lines will ignore `range_end_xpath`.
+#' `column_number_xpath` and `range_start_xpath` must always refer to locations on the starting line of the `xml` node.
```

Comment:
should right? we don't enforce that?

Reply:
Seems impossible, as the `number()` XPath can do anything to arrive.
What we could check is if `start <= column_number <= nchar(line)`, but I'd leave this sanity checking out of this already large PR.

Reply:
oh, not suggesting we enforce it! just changing the doc wording :)

Reply:
ah 😆